File: Tests/CSharp/TestAllDatabases/TestAllDatabases/BusinessObjects/CustomObjects/AggregateDb/FullNameViewCollectionCustom.cs
//======================================
// Copyright EntitySpaces, LLC 2005 - 2006 
//======================================

using System;
using System.ComponentModel;
using System.Collections.Generic;
using System.Text;
using System.Data;

using EntitySpaces.Interfaces;
using EntitySpaces.Core;
using EntitySpaces.DynamicQuery;

namespace BusinessObjects
{
	#region Custom View Tests

	public partial class FullNameViewCollection : esFullNameViewCollection
	{
		public bool CustomLoadFromView()
		{
			string sqlText = "";

			switch (this.es.Connection.ProviderMetadataKey)
			{
				case "esOracle":
					sqlText = "SELECT \"FullName\", \"DepartmentID\", \"HireDate\" ";
					sqlText += "FROM \"MYGENERATION\".\"FullNameView\" ";
					sqlText += "WHERE \"DepartmentID\" = {0} ";
					sqlText += "OR \"DepartmentID\" = {1} ";

                    break;

                case "esPgsql":
                    sqlText = "SELECT \"FullName\", \"DepartmentID\", \"HireDate\" ";
                    sqlText += "FROM \"FullNameView\" ";
                    sqlText += "WHERE \"DepartmentID\" = {0} ";
                    sqlText += "OR \"DepartmentID\" = {1} ";

                    break;

                case "esMySQL":
					sqlText = "SELECT `FullName`, `DepartmentID`, `HireDate` ";
					sqlText += "FROM `FullNameView` ";
					sqlText += "WHERE `DepartmentID` = {0} ";
					sqlText += "OR `DepartmentID` = {1}";

                    break;

				default:
					sqlText = "SELECT [FullName], [DepartmentID], [HireDate] ";
					sqlText += "FROM [FullNameView] ";
					sqlText += "WHERE [DepartmentID] = {0} ";
					sqlText += "OR [DepartmentID] = {1}";

                    break;
            }

            return this.Load(esQueryType.Text, sqlText, 1, 2);
        }

		public bool CustomLoadFromViewNoParams(string whereClause)
		{
			string sqlText = "";

			switch (this.es.Connection.ProviderMetadataKey)
			{
				case "esOracle":
					sqlText = "SELECT \"FullName\", \"DepartmentID\", \"HireDate\" ";
					sqlText += "FROM \"MYGENERATION\".\"FullNameView\" ";
					sqlText += whereClause;

                    break;

                case "esPgsql":
                    sqlText = "SELECT \"FullName\", \"DepartmentID\", \"HireDate\" ";
                    sqlText += "FROM \"FullNameView\" ";
                    sqlText += whereClause;

                    break;

                case "esMySQL":
					sqlText = "SELECT `FullName`, `DepartmentID`, `HireDate` ";
					sqlText += "FROM `FullNameView` ";
					sqlText += whereClause;

                    break;

				default:
					sqlText = "SELECT [FullName], [DepartmentID], [HireDate] ";
					sqlText += "FROM [FullNameView] ";
					sqlText += whereClause;

                    break;
            }

            return this.Load(esQueryType.Text, sqlText);
        }

        public bool CustomLoadFromViewByDept(int deptID)
        {
            this.Query.Where(
                this.Query.DepartmentID == deptID);

            return this.Query.Load();
        }
	}
	#endregion
}

File: Tests/VB/TestAllDatabases/TestAllDatabases/BusinessObjects/CustomObjects/AggregateDb/FullNameViewCollectionCustom.vb
'======================================
' Copyright EntitySpaces, LLC 2005 - 2006 
'======================================

Imports System.ComponentModel
Imports System.Collections.Generic
Imports System.Text
Imports System.Data

Imports EntitySpaces.Interfaces
Imports EntitySpaces.Core
Imports EntitySpaces.DynamicQuery

Namespace BusinessObjects
	#Region "Custom View Tests"

	Public Partial Class FullNameViewCollection
		Inherits esFullNameViewCollection
		Public Function CustomLoadFromView() As Boolean
			Dim sqlText As String = ""

			Select Case Me.es.Connection.ProviderMetadataKey
				Case "esOracle"
					sqlText = "SELECT ""FullName"", ""DepartmentID"", ""HireDate"" "
					sqlText += "FROM ""MYGENERATION"".""FullNameView"" "
					sqlText += "WHERE ""DepartmentID"" = {0} "
					sqlText += "OR ""DepartmentID"" = {1} "

					Exit Select

				Case "esPgsql"
					sqlText = "SELECT ""FullName"", ""DepartmentID"", ""HireDate"" "
					sqlText += "FROM ""FullNameView"" "
					sqlText += "WHERE ""DepartmentID"" = {0} "
					sqlText += "OR ""DepartmentID"" = {1} "

					Exit Select

				Case "esMySQL"
					sqlText = "SELECT `FullName`, `DepartmentID`, `HireDate` "
					sqlText += "FROM `FullNameView` "
					sqlText += "WHERE `DepartmentID` = {0} "
					sqlText += "OR `DepartmentID` = {1}"

					Exit Select
				Case Else

					sqlText = "SELECT [FullName], [DepartmentID], [HireDate] "
					sqlText += "FROM [FullNameView] "
					sqlText += "WHERE [DepartmentID] = {0} "
					sqlText += "OR [DepartmentID] = {1}"

					Exit Select
			End Select

			Return Me.Load(esQueryType.Text, sqlText, 1, 2)
		End Function

		Public Function CustomLoadFromViewNoParams(whereClause As String) As Boolean
			Dim sqlText As String = ""

			Select Case Me.es.Connection.ProviderMetadataKey
				Case "esOracle"
					sqlText = "SELECT ""FullName"", ""DepartmentID"", ""HireDate"" "
					sqlText += "FROM ""MYGENERATION"".""FullNameView"" "
					sqlText += whereClause

					Exit Select

				Case "esPgsql"
					sqlText = "SELECT ""FullName"", ""DepartmentID"", ""HireDate"" "
					sqlText += "FROM ""FullNameView"" "
					sqlText += whereClause

					Exit Select

				Case "esMySQL"
					sqlText = "SELECT `FullName`, `DepartmentID`, `HireDate` "
					sqlText += "FROM `FullNameView` "
					sqlText += whereClause

					Exit Select
				Case Else

					sqlText = "SELECT [FullName], [DepartmentID], [HireDate] "
					sqlText += "FROM [FullNameView] "
					sqlText += whereClause

					Exit Select
			End Select

			Return Me.Load(esQueryType.Text, sqlText)
		End Function

		Public Function CustomLoadFromViewByDept(deptID As Integer) As Boolean
			Me.Query.Where(Me.Query.DepartmentID = deptID)

			Return Me.Query.Load()
		End Function
	End Class
	#End Region
End Namespace
